### PR TITLE
Add flexible commands to MySql modules

### PR DESCRIPTION
**Pull request by DaeunYim**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@DaeunYim Please down load CredScan-Results-matches.sarif from https://dev.azure.com/azure-sdk/internal/_build/results?buildId=627132&view=artifacts&pathAsName=false&type=publishedArtifacts to suppress the credential issues. You can refer to related test cases of MySql.

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please also commit the files in ./examples. The files in ./doc are generated from examples.

**Comment by DaeunYim** [comment]
> Can you provide the autorest cmd you use to generate the code?
It was autorest.cmd --use:@autorest/powershell@3.0.403 <- this command from the email you sent me last time..! Do I need to try with different command?

**Comment by DaeunYim** [comment]
> @DaeunYim Please down load CredScan-Results-matches.sarif from https://dev.azure.com/azure-sdk/internal/_build/results?buildId=627132&view=artifacts&pathAsName=false&type=publishedArtifacts to suppress the credential issues. You can refer to related test cases of MySql.

Resolved this issue...!!

File: src/MySql/docs/Update-AzMySqlVirtualNetworkRule.md [hunk 1]
```diff
@@ -31,30 +31,26 @@ Creates or updates an existing virtual network rule.
 
 ## EXAMPLES
 
-### Example 1: Update MySql Virtual Network Rule by name
```
Comment (msJinLei) [inline src/MySql/docs/Update-AzMySqlVirtualNetworkRule.md:34]: don't find a change in examples/Update-AzMySqlVirtualNetworkRule.md

File: src/MySql/docs/Update-AzMySqlServer.md [hunk 2]
```diff
@@ -39,27 +39,23 @@ Use Update-AzMySqlConfiguration instead if you want update server parameters suc
 
 ## EXAMPLES
 
-### Example 1: Update MySql server by resource group and server name
```
Comment (msJinLei) [inline src/MySql/docs/Update-AzMySqlServer.md:42]: You should commit the change of  examples/Update-AzMySqlServer.md

File: src/MySql/docs/Update-AzMySqlFlexibleServerFirewallRule.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+## DESCRIPTION
+Creates a new firewall rule or updates an existing firewall rule.
+
+## EXAMPLES
+
+### -------------------------- EXAMPLE 1 --------------------------
+```powershell
+Update-AzMySqlFlexibleServerFirewallRule -Name rule -ResourceGroupName PowershellMySqlTest -ServerName mysql-test -EndIPAddress 0.0.0.3 -StartIPAddress 0.0.0.2
+```
```
Comment (msJinLei) [inline src/MySql/docs/Update-AzMySqlFlexibleServerFirewallRule.md:50]: output should be included in the quotation marks

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/MySql/test/New-AzMySqlServer.Tests.ps1 [hunk 4]
```diff
@@ -5,15 +5,18 @@ if (-Not (Test-Path -Path $loadEnvPath)) {
 . ($loadEnvPath)
 $TestRecordingFile = Join-Path $PSScriptRoot 'New-AzMySqlServer.Recording.json'
 $currentPath = $PSScriptRoot
+Write-Output "Inside test PS script222222"
 while(-not $mockingPath) {
     $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
     $currentPath = Split-Path -Path $currentPath -Parent
+    Write-Output "Inside test PS script33333"
 }
```
Comment (msJinLei) [inline src/MySql/test/New-AzMySqlServer.Tests.ps1:13]: invalid output

File: src/MySql/test/New-AzMySqlServer.Tests.ps1 [hunk 5]
```diff
@@ -5,15 +5,18 @@ if (-Not (Test-Path -Path $loadEnvPath)) {
...
 . ($mockingPath | Select-Object -First 1).FullName
 
 Describe 'New-AzMySqlServer' {
     It 'CreateExpanded' {
         {
+            Write-Output "Inside test PS script1111"
```
Comment (msJinLei) [inline src/MySql/test/New-AzMySqlServer.Tests.ps1:19]: invalid output

File: src/MySql/test/utils.ps1 [hunk 6]
```diff
@@ -19,32 +19,34 @@ function setupEnv() {
...
     $resourceGroup = "MySqlTest"
-    $location = "eastus"
+    $location = "westus2"
     $env.Add("resourceGroup", $resourceGroup)
     $env.Add("location", $location)
     New-AzResourceGroup -Name $resourceGroup -Location $location
 
-    # Create the test Vnet
-    write-host "Deploy Vnet template"
-    New-AzDeployment -Mode Incremental -TemplateFile .\test\deployment-templates\virtual-network\template.json -TemplateParameterFile .\test\deployment-templates\virtual-network\parameters.json -Name vn -ResourceGroupName $resourceGroup
```
Comment (msJinLei) [inline src/MySql/test/utils.ps1:34]: why remove the deployment?

**Review by msJinLei — CHANGES_REQUESTED** [review]
I regenerate and commit files under doc. Please take a look at them.

**Comment by msJinLei** [comment]
> > Can you provide the autorest cmd you use to generate the code?
> 
> It was autorest.cmd --use:@autorest/powershell@3.0.403 <- this command from the email you sent me last time..! Do I need to try with different command?

Which powershell version are you using?

File: src/MySql/custom/Update-AzMySqlFlexibleServerConfiguration.ps1 [hunk 7]
```diff
@@ -0,0 +1,125 @@
+function Update-AzMySqlFlexibleServerConfiguration {
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/MySql/custom/Update-AzMySqlFlexibleServerConfiguration.ps1:1]: Added it

File: src/MySql/custom/Update-AzMySqlFlexibleServer.ps1 [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+function Update-AzMySqlFlexibleServer {
+    [OutputType([Microsoft.Azure.PowerShell.Cmdlets.MySql.Models.Api20200701Preview.IServerAutoGenerated])]
+    [CmdletBinding(DefaultParameterSetName='UpdateExpanded', PositionalBinding=$false, SupportsShouldProcess, ConfirmImpact='Medium')]
+    [Microsoft.Azure.PowerShell.Cmdlets.MySql.Description('Updates an existing server. The request body can contain one to many of the properties present in the normal server definition. Use Update-AzMySqlConfiguration instead if you want update server parameters such as wait_timeout or net_retry_count.')]
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/MySql/custom/Update-AzMySqlFlexibleServer.ps1:19]: Updated it. Thanks

File: src/MySql/docs/Update-AzMySqlFlexibleServerFirewallRule.md (hunk 3 above)
Comment (DaeunYim) [reply src/MySql/docs/Update-AzMySqlFlexibleServerFirewallRule.md]: The example files I wrote includes the output in the quotation marks, but the autogenerated docs file don't include it. This was the autogeneration problem I was asking you about.. Not just flexible server documents but also single server documents. Do you think the powershell version would be the problem? I use 7.0.3.

File: src/MySql/test/New-AzMySqlServer.Tests.ps1 [hunk 9]
```diff
@@ -5,15 +5,18 @@ if (-Not (Test-Path -Path $loadEnvPath)) {
 . ($loadEnvPath)
 $TestRecordingFile = Join-Path $PSScriptRoot 'New-AzMySqlServer.Recording.json'
 $currentPath = $PSScriptRoot
+Write-Output "Inside test PS script222222"
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/MySql/test/New-AzMySqlServer.Tests.ps1:8]: Removed

File: src/MySql/test/utils.ps1 (hunk 6 above)
Comment (DaeunYim) [reply src/MySql/test/utils.ps1:34]: Added back

File: src/MySql/docs/Update-AzMySqlServer.md (hunk 2 above)
Comment (DaeunYim) [reply src/MySql/docs/Update-AzMySqlServer.md]: > > > Can you provide the autorest cmd you use to generate the code?
> > 
> > 
> > It was autorest.cmd --use:@autorest/powershell@3.0.403 <- this command from the email you sent me last time..! Do I need to try with different command?
> 
> Which powershell version are you using?

I'm using pwsh 7.0.3 version,,

File: src/MySql/docs/Get-AzMySqlFlexibleServer.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+## DESCRIPTION
+Gets information about a server.
+
+## EXAMPLES
+
+### -------------------------- EXAMPLE 1 --------------------------
+```powershell
+Get-AzMySqlFlexibleServer
+```
+
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/MySql/docs/Get-AzMySqlFlexibleServer.md:45]: The version you autogenerated shows the right output example,,

File: src/MySql/docs/Get-AzMySqlConfiguration.md [hunk 11]
```diff
@@ -34,9 +34,10 @@ Gets information about a configuration of server.
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/MySql/docs/Get-AzMySqlConfiguration.md]: Lei I'm little lost here. I didn't change anything for the single server commands... what kind of commits are you referring here...?

File: src/MySql/docs/Get-AzMySqlConnectionString.md [hunk 12]
```diff
@@ -29,23 +29,19 @@ Get the connection string according to client connection provider.
 
 ## EXAMPLES
 
-### Example 1: Get MySql server connection string by resource group and server name
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/MySql/docs/Get-AzMySqlConnectionString.md]: Same as above

File: src/MySql/docs/Update-AzMySqlVirtualNetworkRule.md (hunk 1 above)
Comment (DaeunYim) [reply src/MySql/docs/Update-AzMySqlVirtualNetworkRule.md]: Same as my reply to the later comment. 

Lei I'm little lost here. I didn't change anything for the single server commands... what kind of commits are you referring here...? Could you kindly explain to me? :)

File: src/MySql/docs/Update-AzMySqlServer.md (hunk 2 above)
Comment (DaeunYim) [reply src/MySql/docs/Update-AzMySqlServer.md]: Same as above..

**Comment by DaeunYim** [comment]
> > > Can you provide the autorest cmd you use to generate the code?
> > 
> > 
> > It was autorest.cmd --use:@autorest/powershell@3.0.403 <- this command from the email you sent me last time..! Do I need to try with different command?
> 
> Which powershell version are you using?

I'm using  PSVersion  5.1.19041.610 and pwsh 7.0.3..

**Review by msJinLei — APPROVED** [review]

File: src/MySql/docs/Update-AzMySqlFlexibleServerFirewallRule.md (hunk 3 above)
Comment (msJinLei) [reply src/MySql/docs/Update-AzMySqlFlexibleServerFirewallRule.md]: > The example files I wrote includes the output in the quotation marks, but the autogenerated docs file don't include it. This was the autogeneration problem I was asking you about.. Not just flexible server documents but also single server documents. Do you think the powershell version would be the problem? I'm using PSVersion 5.1.19041.610 and pwsh 7.0.3..

The commit 19ad16d4ec584fe5791b06b57deb416bcc56dcbe include the codes I generated using your config. I am using the powershell core 7.0.3 but cannot reproduce your issue.

**Pull request merged**